### PR TITLE
PRL-19694 implemented dropDownList extension

### PR DESCRIPTION
**Pull request by shaakem**
screenshot from 2019-02-19 14-33-39

### CODE REVIEW

File: web/datasources/currencyDataSource.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
+export const currencyCodesList = [
```
Comment (DaveyDB) [inline web/datasources/currencyDataSource.js]: These currency codes should come from the DB

Comment (shaakem) [reply web/datasources/currencyDataSource.js]: I'll check with Michael how to implement it.

Comment (michael-penney) [reply web/datasources/currencyDataSource.js]: I think the "powergrid way" is probably to add a new method to the datasource to get the list of options.

Then in drydock, you can make the call to the DB and supply it somehow to the datasource.

File: web/extensions/dropDownList.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/***
+ * dropDownList extension.
+ * Enabled when column type is 'DROP_DOWN_LIST'.
+ * Represents a list of values.
+  */
+
+define(['../override', 'jquery', '../utils', '../datasources/currencyDataSource'], function(override, $, utils, currencyDataSource) {
+
+    const listData = currencyDataSource.currencyCodesList;
```
Comment (michael-penney) [inline web/extensions/dropDownList.js]: should call a function on the datasource to get the list of options.

this.dataSource.getOptions(record, column)

File: web/datasources/currencyDataSource.js (hunk 1 above)
Comment (michael-penney) [reply web/datasources/currencyDataSource.js]: I think the "powergrid way" is probably to add a new method to the datasource to get the list of options.

Then in drydock, you can make the call to the DB and supply it somehow to the datasource.

File: web/extensions/dropDownList.js (hunk 2 above)
Comment (michael-penney) [reply web/extensions/dropDownList.js]: You might also want to add an event-listener, for when the options changed.

File: web/datasources/currencyDataSource.js (hunk 1 above)
Comment (shaakem) [reply web/datasources/currencyDataSource.js]: implemented via extension options

**Comment by ritgup** [comment]
Hi Alexy,

I had a talk with Davey and understood that the look and feel of the dry dock will be changed a bit after merging this ticket.
I will therefore ask you to write a small mail to me and Davey  with screenshots and working after fixing of this ticket so that we can make the customer aware of the changes .

Kind regards,
Ritu Gupta

**Pull request merged**